### ISSUE TITLE
feat: SRC5 Interface detection

### ISSUE THREAD

**Issue by julio4** [NethermindEth/StarknetByExample#109]
Reference: https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-5.md

**Comment by mubarak23** [comment]
@julio4 can you give more context to this

**Comment by julio4** [comment]
> @julio4 can you give more context to this

Show an example of how a contract can implement the `ISRC5` trait to comply with SRC-5 for interfaces detection.

**Comment by mubarak23** [comment]
from the [above docs shared](https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-5.md), we are implementing a trait function that check this `interface_id == 0x3f918d17e5ee77373b56385708f855659a07f75997f365cf87748628532a055`

**Comment by julio4** [comment]
> from the [above docs shared](https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-5.md), we are implementing a trait function that check this `interface_id == 0x3f918d17e5ee77373b56385708f855659a07f75997f365cf87748628532a055`

Not exactly, this specific `interface_id` is only an example of an interface. SRC5 is a standard that define how to generate interface ids and how a contract can publish what interfaces it implements in order to allow anyone to querry if the contract implements an interface by implementing the specific `ISRC5` interface as well.
You can take a look at the OZ implementation for an example [here](https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/introspection/src5.cairo)

**Comment by mubarak23** [comment]
this OpenZeppelin implementation is somewhat same thing we are going to do here

**Comment by julio4** [comment]
> this OpenZeppelin implementation is somewhat same thing we are going to do here

Yes but not as a component (atleast for now), we only want to demonstrate how one contract can implement the trait to return true for a specific interface id. So developers can quickly understand without having to read all the SNIP!

**Comment by mubarak23** [comment]
Alright @julio4 kindly assign this to me, i will work on it

**Comment by Jemiiah** [comment]
i would love to work on this issue @julio4
